cmd/tink-cli/cmd/template/create.go

```diff
@@ -22,54 +22,88 @@ var (
 
 // createCmd represents the create subcommand for template command
 var createCmd = &cobra.Command{
-	Use:     "create",
-	Short:   "create a workflow template ",
-	Example: "tink template create [flags]",
+	Use:   "create",
+	Short: "create a workflow template ",
+	Example: `tink template create [flags]
+cat /tmp/example.tmpl | tink template create -n example`,
+	PreRunE: func(c *cobra.Command, args []string) error {
+		if !isInputFromPipe() {
+			path, _ := c.Flags().GetString(fPath)
+			if path == "" {
+				return fmt.Errorf("either pipe the template or provide the required '--path' flag")
+			}
+		}
+		return nil
+	},
 	Run: func(c *cobra.Command, args []string) {
-		validateTemplate()
-		createTemplate(c, args)
+		var data []byte
+		if isInputFromPipe() {
+			data = readDataFromStdin()
+		} else {
+			data = readTemplateData()
+		}
```

Comment:
What if we reversed this logic a bit, and instead got an io reader:

```
path, _: c.Flags().GetString(fPath)


if path != "" {
  reader, err = os.Open(filePath)
  ...
} else {
  reader = os.Stdin
}
```

and then just `readAll(reader)`? As it stands, it will fairly eagerly read the input from stdin even if I've passed a file.